packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -2695,6 +2709,25 @@ class _InspectorOverlayLayer extends Layer {
   }) {
     return false;
   }
+
+  /// Return whether or not a render object belongs to this inspector widget
+  /// tree.
+  /// The inspector selection is static, so if there are multiple inspector
+  /// overlays in the same app (i.e. an storyboard), a selected or candidate
+  /// render object may not belong to this tree.
+  bool _isInInspectorRenderObjectTree(RenderObject child) {
+    RenderObject inspectorRoot;
+    RenderObject current = child.parent as RenderObject;
+    while (current != null && inspectorRoot == null) {
+      // We found the widget inspector render object.
+      if (current is RenderStack
```

Comment:
This is adding a tight dependency that the _RenderInspectorOverlay has a parent that is a Stack and is the last child of that parent. That is probably ok but is a little fragile. We should either add a key to the Stack that is the parent of the _RenderInspectorOverlay or document in the code where the Stack is used that we are making assumptions about its child.

Reply:
I thought about the key, but I wasn't sure about passing it through arguments all the way down. It looked off. Also the key would have the problem of calling findRenderObject in the first frame right?
I'll add a comment to the Stack at the top as you suggest.

Reply:
as long as it is documented in the build method it is fine as is.